bigbio/hub/hub_repos/jnlpba/jnlpba.py

```diff
@@ -127,55 +138,141 @@ def _info(self) -> datasets.DatasetInfo:
-                gen_kwargs={"data": data["validation"]},
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": test_file},
             ),
         ]
 
-    def _generate_examples(self, data: datasets.Dataset) -> Tuple[int, Dict]:
-        """Yields examples as (key, example) tuples."""
-        uid = 0
+    def _parse_sentence(self, tokens, ner_tags, uid):
+        """
+        This function takes in two stacks, one with tokens and the other with tags
+        It returns the passage and the entities as required by the bigbio_kb schema
+        """
+        entities = []
+        sentence_words = []
+        distance_from_back = -1
+        while tokens:
+            curr_token = tokens.pop()
+            ner_tag = ner_tags.pop()
+            distance_from_back += len(curr_token) + 1
+            sentence_words.append(curr_token)
+            if ner_tag.startswith("I-"):
+                # Keep popping elements until the next B-* tag is hit
+                tag_tokens = [curr_token]
+                curr_tag = ner_tag[2:]
+                while not ner_tag.startswith("B-"):
+                    curr_token = tokens.pop()
+                    ner_tag = ner_tags.pop()
+                    distance_from_back += len(curr_token) + 1
+                    sentence_words.append(curr_token)
+                    tag_tokens.append(curr_token)
+
+                tag_text = " ".join(list(reversed(tag_tokens)))
+                tag_start = distance_from_back
+                tag_end = tag_start - len(tag_text)
+                entity = {
+                    "id": next(uid),
+                    "type": curr_tag,
+                    "text": [tag_text],
+                    "normalized": [],
+                    "offsets": [[tag_start, tag_end]],
+                }
+                entities.append(entity)
+            elif ner_tag.startswith("B-"):
+                curr_tag = ner_tag[2:]
+                tag_start = distance_from_back
+                tag_end = tag_start - len(curr_token)
+                entity = {
+                    "id": next(uid),
+                    "type": curr_tag,
+                    "text": [curr_token],
+                    "normalized": [],
+                    "offsets": [[tag_start, tag_end]],
+                }
+                entities.append(entity)
+            elif ner_tag == "O":
+                continue
+        passage = " ".join(list(reversed(sentence_words)))
+        for entity in entities:
+            entity_start = len(passage) - entity["offsets"][0][0]
+            entity_end = len(passage) - entity["offsets"][0][1]
+            entity["offsets"][0][1] = entity_end
+            entity["offsets"][0][0] = entity_start
+
+        document = {}
+        document["id"] = next(uid)
+        document["document_id"] = document["id"]
+        document["entities"] = entities
+        document["passages"] = [
+            {
+                "id": next(uid),
+                "type": "",
```

Comment:
Passage type should not be empty imho. I guess it is "sentence" in this dataset?